pkg/secrets/k8s/k8s_secrets_test.go

```diff
@@ -41,11 +42,52 @@ func TestKubernetesSecrets(t *testing.T) {
 			})
 		})
 
+		Convey("Given a map of data entries and a secret with backslashes", func() {
+			dataEntries := make(map[string][]byte)
+			// simulates a Conjur secret with 1 backslash with unicode
+			dataEntries["user"] = []byte("super\u005csecret");
+			escapedDataEntry, err := generateStringDataEntry(dataEntries);
+
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("Returns proper password with escaped characters", func() {
+				// returned 2x backslashes because k8s trims the second backslash
+				// to mimic this, we are expecting two backslashes
+				stringDataEntryExpected := `{"stringData":{"user":"super\\secret"}}`
+				stringDataEntryActual := string(escapedDataEntry)
+				eq := reflect.DeepEqual(stringDataEntryActual, stringDataEntryExpected)
+				So(eq, ShouldEqual, true)
+			})
+		})
+
+		// a more complex use-case where the secret contains a combination of backslashes and the unicode escape sequence for backslashes
+		Convey("Given a map of data entries and a secret with unicode and backslashes", func() {
+			dataEntries := make(map[string][]byte)
+			// simulates a Conjur secret with 1 backslash with unicode
+			dataEntries["user"] = []byte("\u005c\u005ca\\u005c\u005cb\u005cc\u005c");
+			escapedDataEntry, err := generateStringDataEntry(dataEntries);
+
+			Convey("Finishes without raising an error", func() {
+				So(err, ShouldEqual, nil)
+			})
+
+			Convey("Returns proper password with escaped characters", func() {
+				// before sending a secret to K8s, each secret will be escaped with an additional backslash
+				// K8s trims the second backslash so in order to mimic this, we are expecting two backslashes
+				stringDataEntryExpected := `{"stringData":{"user":"\\\\a\\u005c\\b\\c\\"}}`
```

Comment:
please count the number of backslashes in the actual and in the expected output and compare them, for the test to be more clear, because it is hard to identify the number of the backslashes in the expected string

Reply:
changes can be seen in new pr in new repo: https://github.com/cyberark/cyberark-secrets-provider-for-k8s/pull/4